### ISSUE TITLE
Cloning an existing backend environment that has notifications with imported auth leads to an unexpected error and dangling resource

### ISSUE THREAD

**Issue by vishal-dms** [aws-amplify/amplify-cli#10244]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.13.2

### Amplify CLI Version

8.0.2 (also exists in previous 7.X version as well)

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

auth, notifications

### Amplify Commands

env

### Describe the bug

On cloning a backend environment (with `amplify env add`) that has Notifications configured with an imported Auth, we face the following error : 
```
✖There was an error initializing your environment.
Error: Auth configuration is required to allow unauthenticated users, but it is not configured properly.
```

### Expected behavior

Cloning the backend environment should succeed without any challenges for this scenario  

### Reproduction steps

1. amplify init (select default configuration)
2. amplify import auth (import a Cognito UserPool that is backed by Identity Pool which has enabled Unauthenticated Access and configured both roles)
3. amplify add notifications (with SMS channel for e.g.) - This should go through successfully if the imported Auth configuration is compatible with Notifications auth requirements. In case this fails, please ensure that the imported Auth resource has correct configuration.
4. amplify push -y (should execute without challenges)
5. amplify env add prod 
```
Note: It is recommended to run this command from the root of your app directory
Using default provider  awscloudformation
? Select the authentication method you want to use: AWS profile

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

? Please choose the profile you want to use default
Adding backend environment prod to AWS Amplify app: <app-ID>
⠙ Initializing project in the cloud...

CREATE_IN_PROGRESS amplify-customtodo-prod-123708 AWS::CloudFormation::Stack Tue Apr 19 2022 12:37:13 GMT+0530 (India Standard Time) User Initiated
CREATE_IN_PROGRESS UnauthRole                     AWS::IAM::Role             Tue Apr 19 2022 12:37:19 GMT+0530 (India Standard Time)               
CREATE_IN_PROGRESS DeploymentBucket               AWS::S3::Bucket            Tue Apr 19 2022 12:37:19 GMT+0530 (India Standard Time)               
CREATE_IN_PROGRESS AuthRole                       AWS::IAM::Role             Tue Apr 19 2022 12:37:19 GMT+0530 (India Standard Time)               
⠙ Initializing project in the cloud...

CREATE_IN_PROGRESS UnauthRole       AWS::IAM::Role  Tue Apr 19 2022 12:37:20 GMT+0530 (India Standard Time) Resource creation Initiated
CREATE_IN_PROGRESS AuthRole         AWS::IAM::Role  Tue Apr 19 2022 12:37:20 GMT+0530 (India Standard Time) Resource creation Initiated
CREATE_IN_PROGRESS DeploymentBucket AWS::S3::Bucket Tue Apr 19 2022 12:37:20 GMT+0530 (India Standard Time) Resource creation Initiated
⠴ Initializing project in the cloud...

CREATE_COMPLETE UnauthRole AWS::IAM::Role Tue Apr 19 2022 12:37:34 GMT+0530 (India Standard Time) 
⠇ Initializing project in the cloud...

CREATE_COMPLETE AuthRole AWS::IAM::Role Tue Apr 19 2022 12:37:34 GMT+0530 (India Standard Time) 
⠙ Initializing project in the cloud...

CREATE_COMPLETE DeploymentBucket               AWS::S3::Bucket            Tue Apr 19 2022 12:37:42 GMT+0530 (India Standard Time) 
CREATE_COMPLETE amplify-customtodo-prod-123456 AWS::CloudFormation::Stack Tue Apr 19 2022 12:37:44 GMT+0530 (India Standard Time) 
✔ Successfully created initial AWS cloud resources for deployments.
✔ Initialized provider successfully.

An Amazon Pinpoint project will be created for notifications.
✔ Successfully created Pinpoint project: <project-name>

✔ Successfully set the IAM policy
✖ There was an error initializing your environment.
Error: Auth configuration is required to allow unauthenticated users, but it is not configured properly.
    at checkAuth (/snapshot/node_modules/amplify-category-notifications/lib/auth-helper.js:116:11)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at Object.ensureAuth (/snapshot/node_modules/amplify-category-notifications/lib/auth-helper.js:19:3)
    at createPinpointApp (/snapshot/node_modules/amplify-category-notifications/lib/pinpoint-helper.js:118:3)
    at Object.ensurePinpointApp (/snapshot/node_modules/amplify-category-notifications/lib/pinpoint-helper.js:65:20)
    at pushChanges (/snapshot/node_modules/amplify-category-notifications/lib/multi-env-manager.js:214:3)
    at Object.initEnv (/snapshot/node_modules/amplify-category-notifications/lib/multi-env-manager.js:14:5)
    at initEnv (/snapshot/node_modules/amplify-category-notifications/index.js:16:3)


$ amplify status

    Current Environment: prod
    
┌───────────────┬────────────────────┬───────────┬───────────────────┐
│ Category      │ Resource name      │ Operation │ Provider plugin   │
├───────────────┼────────────────────┼───────────┼───────────────────┤
│ Notifications │ CustomTodo5zBMl    │ Create    │                   │
├───────────────┼────────────────────┼───────────┼───────────────────┤
│ Auth          │ customtodo2d1967b7 │ Import    │ awscloudformation │
└───────────────┴────────────────────┴───────────┴───────────────────┘
```

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

The original environment did not face any challenges with the imported Auth resource, we expect the Cloned environment should provision successfully. Additionally we also do not see a prompt asking whether we actually need the imported resource in the new environment as well. Lastly, post this error in the new environment, we are not able to interact (modify/remove) the Notifications resource that failed to create and it seems to be a dangling resource.

**Comment by josefaidt** [comment]
Hey @vishal-dms :wave: thanks for raising this! I was able to successfully reproduce this issue using the provided steps. As a note, in the original environment if I were to import an auth resource that did not allow unauthenticated logins I would receive this error upon adding the notification resource with `amplify env add` > `SMS`, however after importing a resource that does allow unauth the notification was created successfully and we receive this error on `amplify env add` as you have noted.

Marking as a bug 🙂

**Comment by josefaidt** [comment]
Hey @vishal-dms as a workaround, can you try copying the `categories.auth` block for the existing environment of the `amplify/team-provider-info.json` file to the new environment, then re-attempt adding the environment? After the failed attempt, we should see a new block in that file for the environment. For example:

```json
{
  "dev": {
    "awscloudformation": { /* ... */ },
    "categories": {
      "auth": {
        "1024470628189": { /* ... */ }
      },
      "notifications": { /* ... */ }
    }
  },
  "mynewenv": {
    "awscloudformation": { /* ... */ },
  }
}
```

Note in the example above we have the new environment name from the failed attempt with a `awscloudformation` block. If we are wanting to use the same auth resource for the new environment we can copy the `categories.auth` block of the existing environment to the new:

```diff
{
  "dev": {
    "awscloudformation": { /* ... */ },
    "categories": {
      "auth": {
        "1024470628189": { /* ... */ }
      },
      "notifications": { /* ... */ }
    }
  },
  "mynewenv": {
    "awscloudformation": { /* ... */ },
+   "categories": {
+     "auth": {
+       "1024470628189": { /* ... */ }
+     }
+   }
  }
}
```

Finally, upon running `amplify env add mynewenv` and stepping through the prompts we can successfully initialize the new environment.

**Comment by lazpavel** [comment]
This issue is fixed in latest Amplify CLI (10.8.0)
